### ISSUE TITLE
Decimals are not filtering or ordering as expected.

### ISSUE THREAD

**Issue by shayasmk1** [meilisearch/meilisearch#4339]
Decimals when used in a document and uploaded to index, this does not seems to either filter or order correctly. When tried to order decimals in ascending order, some prices comes inbetween. 

Also, when <,> are used with filters it does not work and the final result will have no records. 

For example, 

a filter condition like this
We are only trying to do it from the server. We are filtering this like 

supplier_price > 0 AND supplier_price < 10000
result in 0 results. 

![IMG-20240118-WA0001](https://github.com/meilisearch/meilisearch/assets/12490579/22194e2b-5435-44e0-bed7-e5570b88b006)

**Comment by shayasmk1** [comment]
![Uploading IMG-20240118-WA0001.jpeg…]()

**Comment by irevoire** [comment]
Hey @shayasmk1, 

Can you share more information about your bug, like the structure of your documents and a specific case where it doesn't work so we can reproduce the issue?

On my side, I did a simple query and couldn't reproduce it:
```json
% echo '{ "filter": "a > 1 AND a < 10"}' | mieli search
{
  "hits": [
    {
      "id": 15,
      "a": 1.12
    },
    {
      "id": 16,
      "a": 4
    }
  ],
  "query": "",
  "processingTimeMs": 0,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 2
}
```

**Comment by vhar** [comment]
Same problem when comparing numeric values

**Comment by irevoire** [comment]
Hey, I still can't reproduce, sadly.

I really need someone to show me step-by-step how to reproduce your issue otherwise, I can't do anything about it :pensive:

**Comment by Kerollmops** [comment]
Hey @shayasmk1 and welcome back @vhar 👋

Your issue is probably related to a bug. Unfortunately, it seems that similar issues were reported on https://github.com/meilisearch/meilisearch/discussions/4390 and https://github.com/meilisearch/meilisearch/issues/4374. We will investigate this next week.

However, to investigate, we need a complete reproduction example. What's the version of Meilisearch you use? Do you have a script or something to send your settings and documents to Meilisearch? It's important to send them in batches and not give us only the documents, as it is probably related to updating the documents during the lifetime of the Meilisearch index.

Have a good day 🌻

**Comment by shayasmk1** [comment]
Hello,

What we are doing is just this.

I am using php. SO, the code will be in php. LIke this

$index->search($search,
                ['limit' => 15, 'offset' => ($page - 1) * 15,
                    'sort' => ['created_at:desc'],
                    'filter' => ['subcategory = 40 AND supplier_price
> 0 AND supplier_price < 100000']
                    ],
                )->getRaw();

If giving 'subcategory = 40 only, does not have any issues. But when
we try to add supplier_price, it starts to give empty results.

supplier_price is decimal and subcategory is integer.

On 2/7/24, Clément Renault ***@***.***> wrote:
> Hey @shayasmk1 and welcome back @vhar 👋
>
> Your issue is probably related to a bug. Unfortunately, it seems that
> similar issues were reported on
> https://github.com/meilisearch/meilisearch/discussions/4390 and
> https://github.com/meilisearch/meilisearch/issues/4374. We will investigate
> this next week.
>
> However, to investigate, we need a complete reproduction example. What's the
> version of Meilisearch you use? Do you have a script or something to send
> your settings and documents to Meilisearch? It's important to send them in
> batches and not give us only the documents, as it is probably related to
> updating the documents during the lifetime of the Meilisearch index.
>
> Have a good day 🌻
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/meilisearch/meilisearch/issues/4339#issuecomment-1932359024
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
Mohammed Shayas M K

**Comment by shayasmk1** [comment]
Meliesearch Version 1.6.1

On 2/10/24, Mohammed shayas M K ***@***.***> wrote:
> Hello,
>
> What we are doing is just this.
>
> I am using php. SO, the code will be in php. LIke this
>
> $index->search($search,
>                 ['limit' => 15, 'offset' => ($page - 1) * 15,
>                     'sort' => ['created_at:desc'],
>                     'filter' => ['subcategory = 40 AND supplier_price
>> 0 AND supplier_price < 100000']
>                     ],
>                 )->getRaw();
>
> If giving 'subcategory = 40 only, does not have any issues. But when
> we try to add supplier_price, it starts to give empty results.
>
> supplier_price is decimal and subcategory is integer.
>
> On 2/7/24, Clément Renault ***@***.***> wrote:
>> Hey @shayasmk1 and welcome back @vhar 👋
>>
>> Your issue is probably related to a bug. Unfortunately, it seems that
>> similar issues were reported on
>> https://github.com/meilisearch/meilisearch/discussions/4390 and
>> https://github.com/meilisearch/meilisearch/issues/4374. We will
>> investigate
>> this next week.
>>
>> However, to investigate, we need a complete reproduction example. What's
>> the
>> version of Meilisearch you use? Do you have a script or something to send
>> your settings and documents to Meilisearch? It's important to send them
>> in
>> batches and not give us only the documents, as it is probably related to
>> updating the documents during the lifetime of the Meilisearch index.
>>
>> Have a good day 🌻
>>
>> --
>> Reply to this email directly or view it on GitHub:
>> https://github.com/meilisearch/meilisearch/issues/4339#issuecomment-1932359024
>> You are receiving this because you were mentioned.
>>
>> Message ID: ***@***.***>
>
>
> --
> Mohammed Shayas M K
>


-- 
Mohammed Shayas M K

**Comment by Kerollmops** [comment]
Hello @shayasmk1 and @vhar 👋

Thank you for your answer, but it doesn't help much more. We need your dataset or some help on what's in it to be able to reproduce. Maybe @vhar can help with this? Can you please share both your dataset and the query you were performing? We have already tried to reproduce but have been unable to.

Have a good day.

**Comment by shayasmk1** [comment]
I can give you sample which we are using to upload.

Array ( [0] => Array ( [id] => 702616 [name] => dsflk; [type] =>
General [created_at] => 2024-01-31 00:28:00 [manufacturer] =>
[supplier_price] => 12.00 [slug] => dsflk- [image] =>
150560_310124002736_loader.png [image_location] => Server [category]
=> 49 [subcategory] => 162 [brand] => 12 [stock] => 1 ) [1] => Array (
[id] => 121900 [name] => Holzmann TKS500S Construction and Table Saw
400V [type] => General [created_at] => 2024-01-16 22:34:51
[manufacturer] => [supplier_price] => 657.58 [slug] =>
Holzmann-TKS500S-Construction-and-Table-Saw-400V [image] =>
[image_location] => Server [category] => 50 [subcategory] => 232
[brand] => 1291 [stock] => 1 ) )

This is a set we are uploading.

THis is assigned to $productArray
Now,
What we do is
$index = $client->index('products1');
    $index->addDocuments($productArray);

$index->updateSortableAttributes([
        'created_at',
        'supplier_price'

      ]);

    $index->updateFilterableAttributes([
        'subcategory',
        'category',
        'brand',
        'name',
        'supplier_price'
        ]);

On 2/12/24, Clément Renault ***@***.***> wrote:
> Hello @shayasmk1 and @vhar 👋
>
> Thank you for your answer, but it doesn't help much more. We need your
> dataset or some help on what's in it to be able to reproduce. Maybe @vhar
> can help with this? Can you please share both your dataset and the query you
> were performing? We have already tried to reproduce but have been unable
> to.
>
> Have a good day.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/meilisearch/meilisearch/issues/4339#issuecomment-1938832242
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
Mohammed Shayas M K

**Comment by Kerollmops** [comment]
Hey @shayasmk1 👋

Good news: We found the root cause of the issue and will land a v1.6.2 with a hotfix, which will be fixed in v1.7 cleaner. The bug is in the incremental part of the facets, and we removed it for the hotfix but will fix it correctly in v1.7.

Closing this in favor of #4409.

Thank you again for your report ❤️
